test/end-to-end/data.test.ts

```diff
@@ -333,6 +333,69 @@ describe('X-AR-IO-Root-Transaction-Id header', function () {
   });
 });
 
+describe('X-AR-IO-Data-Item-Data-Offset header', function () {
+  let compose: StartedDockerComposeEnvironment;
+  const bundle = '-H3KW7RKTXMg5Miq2jHx36OHSVsXBSYuE2kxgsFj6OQ';
+  const bdi = 'fLxHz2WbpNFL7x1HrOyUlsAVHYaKSyj6IqgCJlFuv9g';
+  const di = 'Dc-q5iChuRWcsjVBFstEqmLTx4SWkGZxcVO9OTEGjkQ';
+
+  before(async function () {
+    await rimraf(`${projectRootPath}/data/sqlite/*.db*`, { glob: true });
+
+    compose = await new DockerComposeEnvironment(
+      projectRootPath,
+      'docker-compose.yaml',
+    )
+      .withEnvironment({
+        START_HEIGHT: '1',
+        STOP_HEIGHT: '1',
+        GET_DATA_CIRCUIT_BREAKER_TIMEOUT_MS: '100000',
+        ANS104_UNBUNDLE_FILTER: '{"always": true}',
+        ANS104_INDEX_FILTER: '{"always": true}',
+        ADMIN_API_KEY: 'secret',
+      })
+      .withBuild()
+      .withWaitStrategy('core-1', Wait.forHttp('/ar-io/info', 4000))
+      .up(['core']);
+
+    await axios.post(
+      'http://localhost:4000/ar-io/admin/queue-bundle',
+      { id: bundle },
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: 'Bearer secret',
+        },
+      },
+    );
+
+    // hopefully enough time to unbundle it
+    await wait(10000);
+  });
+
+  after(async function () {
+    await compose.down();
+  });
+
+  it('Verifying header for L1 bundle', async function () {
```

Comment:
@djwhitt do you think I should query DB to check if the calculation was done properly?
I don't think it make sense as this bundle will never change. One thing that I could do is to describe the reasoning like:
bdi has `root_parent_offset` X and `data_offset` Y so data item data offset should be Z.

Reply:
@karlprieb Let's skip checking against the DB, but add a comparison of the data retrieved by requesting `Dc-q5iChuRWcsjVBFstEqmLTx4SWkGZxcVO9OTEGjkQ` by ID and using the size + offset. That seems like a good sanity check since that's the functionality we're ultimately trying to enable.

Reply:
@djwhitt here [`507bc89` (#228)](https://github.com/ar-io/ar-io-node/pull/228/commits/507bc892eb7d70c2edd6193976618b22c68daa8c)